plugins/ipam/dhcp/lease.go

```diff
@@ -292,6 +292,11 @@ func (l *DHCPLease) maintain() {
 	for {
 		var sleepDur time.Duration
 
+		if _, err := netlink.LinkByName(l.link.Attrs().Name); err != nil {
+			log.Printf("%v: interface %s no longer exists, terminating lease maintenance", l.clientID, l.link.Attrs().Name)
+			return
```

Comment:
I was going to release at this point, but, in my testing, if the interface no longer exists -- the call to release fails.

Reply:
We had a problem in Cilium with LinkByName returning ErrDumpInterrupted on churny nodes. I wonder if we should wrap that here in a retry too.